src/bp-core/bp-core-functions.php

```diff
@@ -8962,3 +8965,91 @@ function bb_is_allowed_register_email_address( $email = '' ) {
 		return $is_allowed;
 	}
 }
+
+/**
+ * Wrapper function to get the redirect url as per action.
+ *
+ * @since BuddyBoss [BBVERSION]
+ *
+ * @param string $redirect_to Original redirect URL.
+ * @param int    $user_id     User ID.
+ * @param string $action      Type of action i.e. login or logout.
+ *
+ * @return string $redirect_to Updated redirect URL.
+ */
+function bb_redirect_after_action( $redirect_to, $user_id = 0, $action = 'login' ) {
+
+	$redirect_setting = '';
+
+	if ( 'login' === $action ) {
+		$redirect_setting = bb_login_redirection();
+	} elseif ( 'logout' === $action ) {
+		$redirect_setting = bb_logout_redirection();
+	}
+
+	// Check if any page or custom URL is set.
+	if ( '' !== $redirect_setting ) {
+		if ( '0' === $redirect_setting ) {
+
+			if ( 'login' === $action ) {
+				$custom_url = esc_url( bb_custom_login_redirection() );
+			} elseif ( 'logout' === $action ) {
+				$custom_url = esc_url( bb_custom_logout_redirection() );
+			}
+
+			if ( ! empty( $custom_url ) ) {
+
+				// Custom Page URL.
+				$redirect_to = $custom_url;
+			}
+		} else {
+
+			// Page ID.
+			if (
+				! empty( $redirect_setting ) &&
+				is_numeric( $redirect_setting ) &&
+				'publish' === get_post_status( $redirect_setting )
+			) {
+				$redirect_to = get_permalink( $redirect_setting );
+			}
+		}
+	}
+
+	// Check for profile type settings for login redirection.
+	if ( false !== bp_member_type_enable_disable() ) {
+		$member_type = bp_get_member_type( $user_id );
+		if ( false !== $member_type ) {
+			$member_type_post_id = bp_member_type_post_by_type( $member_type );
+			if ( ! empty( $member_type_post_id ) ) {
+
+				$member_type_redirect_setting = get_post_meta( $member_type_post_id, '_bp_member_type_' . $action . '_redirection', true );
+				// Check if any page or custom URL is set.
+				if ( '' !== $member_type_redirect_setting ) {
+					if ( '0' === $member_type_redirect_setting ) {
+						$custom_url = get_post_meta( $member_type_post_id, '_bp_member_type_custom_' . $action . '_redirection', true );
+						if ( ! empty( $custom_url ) ) {
+
+							// Custom Page URL.
+							$redirect_to = $custom_url;
+						}
+					} else {
+
+						// Page ID.
+						if (
+							! empty( $member_type_redirect_setting ) &&
+							is_numeric( $member_type_redirect_setting ) &&
+							'publish' === get_post_status( $member_type_redirect_setting )
+						) {
+							$redirect_to = get_permalink( $member_type_redirect_setting );
+						}
+					}
+				}
+			}
+		}
+	}
+
+	// Support for custom third party URLs.
+	add_filter( 'allowed_redirect_hosts', 'bb_redirection_allowed_third_party_domains' );
```

Comment:
Make this hook conditional when the custom URL is selected and then the filter will gonna apply.